src/libraries/System.Private.CoreLib/src/System/Array.cs

```diff
@@ -1722,7 +1722,8 @@ public static void Reverse<T>(T[] array)
         {
             if (array == null)
                 ThrowHelper.ThrowArgumentNullException(ExceptionArgument.array);
-            Reverse(array, 0, array.Length);
+            if (array.Length > 1)
+                SpanHelpers.Reverse(ref MemoryMarshal.GetArrayDataReference(array), (nuint)array.Length);
```

Comment:
before we merge this change, we need to update our tests, as some of them are using `Array.Reverse` to get the expected output for `Span.Reverse`:

https://github.com/dotnet/runtime/blob/c663fa429a88d3089b740a89181c2b6f033b2839/src/libraries/System.Memory/tests/Span/Reverse.cs#L134-L137

I am going to send a PR for that in a few minutes.

Reply:
Done: https://github.com/dotnet/runtime/pull/68493

Reply:
Good catch, I didn't notice that. Sounds like a good idea to wait on this PR until the tests are updated, just to be sure.

Reply:
Why can't this just be `ref array[0]`?